### ISSUE TITLE
device.wake() toggles screen

### ISSUE THREAD

**Issue by nickwender** [dtmilano/AndroidViewClient#47]
With adbclient, the `device.wake()` command presses the power button, toggling the sleep/wake state of the screen. If the screen is awake already and you issue `device.wake()` the screen will go to sleep.

This is an issue because I often want to ensure that the screen is awake prior to taking a screenshot (otherwise, you get a blank screenshot).  As is, I don't think there is a method to determine if the screen is awake.

I have two thoughts on this:

(1)

Perhaps look into sending the wake event like ChimpManager does (http://androidxref.com/4.3_r2.1/xref/tools/swt/chimpchat/src/main/java/com/android/chimpchat/ChimpManager.java#376).  This approach ultimately writes "wake\n" to a socket connection to the monkey daemon on the device.  Looking through AndroidViewClient's AdbClient script, does this talk to only ADB on the host (the PC the script runs on) or does AdbClient also talk to the device's monkey daemon directly?  Talking to the monkey daemon directly may be valuable.  I think I have some proof-of-concept code for this if you want it.

(2)

This method could use `adb shell dumpsys power` and look at the `mWakefulness` attribute.  This may vary from device to device, but on my Nexus 7 it returns "Asleep" if the device is asleep and "Awake" when the screen is on already.

**Comment by JaredByers** [comment]
I had a similar problem and was able to solve it using the below script. The device I was using was set to swipe to unlock, hence the "swipe" action in there. But you can change that depending on the type of lock you have set or remove it all together if you don't have the lock screen enabled.

```
result="$(adb shell dumpsys input_method | grep -c "mScreenOn=true")"

if [ "$result" == 1 ]; then
	echo "Screen is already on."

else
	echo "Turning screen on."
	adb shell input keyevent 26
	sleep 2
	adb shell input swipe 900 1100 200 1000 150
fi
```

Hope that helps! Cheers.

**Comment by dtmilano** [comment]
Is `AdbClient.isScreenOn()` returning a wrong value in your case?
If yes, what's the device, android version, etc.

**Comment by jbyers92** [comment]
No, I didn't even use that. I just meant I had faced a problem of trying to get the screen's status and then performing some action based on it. This is the script that worked for me. Sorry if I misunderstood the question.

**Comment by infozsoda** [comment]
You may want to use `input keyevent KEYCODE_WAKEUP`
